include/kframework/foundry.md

```diff
@@ -1,71 +1,86 @@
+
+The Foundry integration allows users to take Solidity property tests and generate K specifications which can be executed using the Haskell symbolic backend.
+
+Before executing any of the KEVM instructions, make sure that you have the following:
+   1. Successfully built or installed KEVM,
+   2. The`kevm` binary is on your PATH,
+   3. Activated the virtual environment (*applicable only for builds from source*).
+
+Below we are providing an example usage and a description of all the commands you can use with KEVM to improve your experience.
+
+Available Commands
+------------------
+
+Basic commands are (and each can be passed --help):
+
+- `kevm foundry-kompile`: Kompile a definition, generating claims for each Foundry test.
+The best options are:
+   - `--regen` - needed if Solidity sources change,
+   - `-rekompile` - needed if K lemmas change, or K definition changes,
+   - `--require` - for adding an extra K file with lemmas, 
+   - `--module-import` - importing an extra K module in one of the added K files with lemmas.
+
+- `kevm foundry-prove`: Run a given proof using the KCFG-based prover (not supporting loops yet, need to fall back to typical K for that).
+The best options are:
+   - `—reinit` - want to start over from scratch,
+   - `—no-simplify-init` - do not want to invoke simplification on all the original nodes, can be faster,
+   - `—max-depth` - increase the space between saved nodes; bigger is faster,
+   - `—max-iterations` - maximum number of nodes to store in KCFG before giving on attempting proof for that run,
+   - `—break-every-step` - save a state every opcode, slow, good for debugging,
+   - `—break-on-calls` - save a state every time a call is made, good to turn on.
+
+- `kevm foundry-show`: Display the given KCFG so far as text. 
+Options are:
+   - `—no-minimize` - do not omit all the gory details, 
+   - `—node` - can be a repeated option, display more information about a given node   hash,
+   - `—node-delta` - displays the delta between two given nodes in the KCFG.
+
+- `kevm foundry-view-kcfg`: Launch the more interactive exploration of the KCFG (can be done while exploration is running, must Ctrl-C + relaunch to view updates to KCFG).
+
+- `kevm view-kcfg [save_directory] [spec_file] [--claim claim_label] ...` command takes the same basic arguments as `kevm prove --pyk ...` does, including:
+  - `save_directory` must be passed as where the KCFGs have been saved (by a previous call to `kevm prove --pyk --save-directory save_directory ...`
+  - `spec_file` is the file to look in for specifications. This file is read the same way as with `kevm prove —pyk …`; the `KProve.get_claims` is called, invoking the frontend.
+  - `--claim claim_label` option is added, but unlike `kevm prove --pyk ...`, you can only repeat it once. This option lets you select an individual claim out of the `spec_file`; if not supplied, it’s assumed that only one spec is present.
+  - `--spec-module spec_module` is also an inherited option.
```

Comment:
This one is not a foundry option (neither are the code blocks following this), so should not be in this file.